### PR TITLE
docs: edit intro testing page

### PR DESCRIPTION
**Pull request by kapunahelewong**
This commit edits the copy of the introductory page to the testing section. This includes moving the CI
sections into the Tutorials. It removes colloquialisms and brings document in line with Google styleguide in intro and CI sections.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Currently text isn't completely inline with Google style guide.
 
Issue Number: N/A

## What is the new behavior?

Updates to style guide styles.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/testing.md [hunk 1]
```diff
@@ -57,237 +55,66 @@ Chrome ...: Executed 3 of 3 SUCCESS (0.135 secs / 0.205 secs)
...
-</code-example>
-
-<div class="alert is-helpful">
-
-   **Note:** Right now, you'll also want to include the `--disable-gpu` flag if you're running on Windows. See [crbug.com/737678](https://crbug.com/737678).
-
-</div>
+Though the file extension `.spec.ts` is a requirement, the location of the tests being in the same directory is the component that they test is a convention.
 
+This convention makes tests easy to find, helps reveal how a component works in context, and reminds you to include the tests when moving or renaming parts of your application.
```
Comment (LayZeeDK) [inline aio/content/guide/testing.md:104]: Is this comma correct and needed?

```suggestion
This convention makes tests easy to find, helps reveal how a component works in context and reminds you to include the tests when moving or renaming parts of your application.
```

File: aio/content/guide/testing.md [hunk 2]
```diff
@@ -57,237 +55,66 @@ Chrome ...: Executed 3 of 3 SUCCESS (0.135 secs / 0.205 secs)
...
+<div class="callout is-helpful">
 
-The CLI generated a test file for the `AppComponent` named `app.component.spec.ts`.
+<header>Using other test frameworks</header>
 
-<div class="alert is-important">
-
-The test file extension **must be `.spec.ts`** so that tooling can identify it as a file with tests (AKA, a _spec_ file).
+Though the CLI uses the Jasmine test framework by default, you can unit test an Angular application with other testing libraries and test runners.
+This guide specifically covers Jasmine and Karma, so if you use another option, be sure to research its distinctive installation procedures, configuration, and syntax.
```
Comment (LayZeeDK) [inline aio/content/guide/testing.md:83]: More formal suggestion.

```suggestion
This guide specifically covers Jasmine and Karma. If you use another option, be sure to research its distinctive installation procedures, configuration, and syntax.
```

File: aio/content/guide/testing.md [hunk 3]
```diff
@@ -57,237 +55,66 @@ Chrome ...: Executed 3 of 3 SUCCESS (0.135 secs / 0.205 secs)
...
-You can click on a test row to re-run just that test or click on a description to re-run the tests in the selected test group ("test suite").
+You can click on a test row to re-run just that test or click on a description to re-run the tests in the selected test group, or "test suite".
 
 Meanwhile, the `ng test` command is watching for changes.
 
 To see this in action, make a small change to `app.component.ts` and save.
 The tests run again, the browser refreshes, and the new test results appear.
 
-## Configuration
+### Editing test configuration
```
Comment (LayZeeDK) [inline aio/content/guide/testing.md:71]: Did you mean to adjust the heading level?

```suggestion
## Editing test configuration
```

Are the headings following this one aligned?

**Review by LayZeeDK — COMMENTED** [review]
Nice improvements 👍

File: aio/content/guide/testing.md (hunk 1 above)
Comment (kapunahelewong) [reply aio/content/guide/testing.md:104]: Hi @LayZeeDK! You're right! I was thinking of the second part as an independent clause but it's missing the subject. Nice catch. Thanks :)

File: aio/content/guide/testing.md [hunk 4]
```diff
@@ -57,237 +55,66 @@ Chrome ...: Executed 3 of 3 SUCCESS (0.135 secs / 0.205 secs)
...
 
 <hr />
 
 ## More info on testing
 
 After you've set up your app for testing, you may find the following testing  guides useful.
 
-* [Code coverage](guide/testing-code-coverage)&mdash;find out how much of your app your tests are covering and how to specify required amounts.
-* [Testing services](guide/testing-services)&mdash;learn how to test the services your app uses.
 * [Basics of testing components](guide/testing-components-basics)&mdash;discover the basics of testing Angular components.
```
Comment (jbogarthyde) [inline aio/content/guide/testing.md:112]: I assume you are planning to arrange the subpages in this order in the TOC, too.

Comment (jbogarthyde) [inline aio/content/guide/testing.md:112]: "discover",  "learn how", "read about", "find out", "uncover" --Should probably standardize descriptions on whatever term you choose.

Perhaps: 
 [Testing components](guide/testing-components-basics): Learn the basics of testing Angular components.

**Review by jbogarthyde — COMMENTED** [review]
I know it's early days - a few suggestions.

File: aio/content/guide/testing.md (hunk 3 above)
Comment (kapunahelewong) [reply aio/content/guide/testing.md:71]: Yes, I did mean to change the header level. The assumption in this section is that if you're using the CLI you already have Jasmine Karma configured by default as discussed just above this section. Is it confusing? What do you think? It's still very much a draft mode PR and there's a lot of flexibility. :)

File: aio/content/guide/testing.md [hunk 5]
```diff
@@ -6,17 +6,17 @@ Testing your Angular application helps you check that your app is working as you
...
 * HTML
 * CSS
+* JavaScript
+* [Angular fundamentals](guide/architecture)
 * [Angular CLI](/cli)
 
 <hr>
 
-The testing documentation offers tips and techniques for unit and integration testing Angular applications through a sample application created with the [Angular CLI](cli).
+The testing documentation offers tips and techniques for testing Angular applications through a sample application created with the [Angular CLI](cli).
```
[the remark this answers is not in the document]
Comment (kapunahelewong) [reply aio/content/guide/testing.md:19]: Nope, we decided to get rid of that on Friday and I forgot to push up. And regarding "unit testing" versus "integration testing", etc., it's been really difficult to find external resources for these that are appropriate. If you see anything let me know!

File: aio/content/guide/testing.md (hunk 4 above)
Comment (kapunahelewong) [reply aio/content/guide/testing.md:112]: That is correct. We're currently experimenting with the order.

File: aio/content/guide/testing.md (hunk 3 above)
Comment (LayZeeDK) [reply aio/content/guide/testing.md]: You are in a better position to decide this, so let's follow your intuition, @kapunahelewong.

Comment (kapunahelewong) [reply aio/content/guide/testing.md]: Thanks, @LayZeeDK. :) I thought of you though when I decided to remove some of the intro stuff and bumped this back up to an `h2`! You were right all along! 🌼

**Review by josephperrott — APPROVED** [review]
LGTM, Reviewed-For:code-ownership

File: aio/content/guide/continuous-integration.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
+# Setting up continuous integration
+
+Continuous integration (CI) servers let you set up your project repository so that your tests run on every commit and pull request, which helps minimize bugs in your project.
+
+There are paid CI services like Circle CI and Travis CI, or you can host your own for free using services such as Jenkins.
+Although Circle CI and Travis CI are paid services, they are free for open source projects.
+You can create a public project on GitHub and add these services without paying.
```
Comment (aikidave) [inline aio/content/guide/continuous-integration.md:7]:
```suggestion
You can select one of several CI services. Popular choices include Circle CI and Travis CI, which are paid services that have a free option for open source projects. You can also host your own CI using services such as Jenkins.
```

File: aio/content/guide/continuous-integration.md [hunk 7]
```diff
@@ -0,0 +1,144 @@
+# Setting up continuous integration
+
+Continuous integration (CI) servers let you set up your project repository so that your tests run on every commit and pull request, which helps minimize bugs in your project.
```
Comment (aikidave) [inline aio/content/guide/continuous-integration.md:3]: Continuous integration (CI) *servers*, or *services*?

File: aio/content/guide/continuous-integration.md [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+There are paid CI services like Circle CI and Travis CI, or you can host your own for free using services such as Jenkins.
+Although Circle CI and Travis CI are paid services, they are free for open source projects.
+You can create a public project on GitHub and add these services without paying.
+
+Contributions to the Angular repo automatically run through a whole suite of Circle CI tests.
```
Comment (aikidave) [inline aio/content/guide/continuous-integration.md:9]: Is this sentence necessary? Are we explaining how to contribute code to Angular, or how users can test their own code? If the latter, I don't think this sentence is required.

File: aio/content/guide/continuous-integration.md [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+1. Commit your changes and push them to your repository on GitHub.
+
+1. [Sign up for Circle CI](https://circleci.com/docs/2.0/first-steps/#section=getting-started) and add your project.
+Your project should start building.
+
+For more information, see the [Circle CI documentation](https://circleci.com/docs/2.0/).
+
+## Configuring Travis CI
+
+Travis CI automatically builds and tests changes you make to your code and lets you know if those changes are successful.
```
Comment (aikidave) [inline aio/content/guide/continuous-integration.md:51]:
```suggestion
Travis CI automatically builds and tests the changes you make to your code and lets you know if those changes are successful.
```

File: aio/content/guide/continuous-integration.md [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+
+  install:
+    - npm install
+
+  script:
+    - npm run test -- --no-watch --no-progress --browsers=ChromeHeadlessCI
+    - npm run e2e -- --protractor-config=e2e/protractor-ci.conf.js
+  ```
+
+  This configuration is almost the same as the Circle CI configuration.
```
Comment (aikidave) [inline aio/content/guide/continuous-integration.md:82]: Line 82 (and maybe 83) aren't necessary. They assume that users have read the previous section, but they might not have. They're just as likely to skip to this section because they want to use Travis CI.

File: aio/content/guide/continuous-integration.md [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+
+## Configuring the CLI for CI testing in Chrome
+
+You can  adjust your configuration to run the Chrome browser tests when running the Angular CLI commands `ng test` and `ng e2e`.
+
+There are configuration files for both the [Karma JavaScript test runner](https://karma-runner.github.io/latest/config/configuration-file.html)
+and [Protractor](https://www.protractortest.org/#/api-overview) end-to-end testing tool,
+which you must adjust to start Chrome without sandboxing.
+
+<!-- What does sandboxing mean here? And does this mean to be starting chrome automatically as the goal? -->
```
Comment (aikidave) [inline aio/content/guide/continuous-integration.md:100]: Should this comment be addressed before we merge this PR?

Comment (LayZeeDK) [reply aio/content/guide/continuous-integration.md:93]: To try to answer the question, it's about adding the `--no-sandbox` option as seen below.

This is an easy way to tell Chrome to trust all content. The alternative is to set up a Chrome sandbox as described here: https://developers.google.com/web/tools/puppeteer/troubleshooting#setting_up_chrome_linux_sandbox

File: aio/content/guide/continuous-integration.md [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+
+For more information about Travis CI testing, see the [Travis CI documentation](https://docs.travis-ci.com/).
+
+## Configuring the CLI for CI testing in Chrome
+
+You can  adjust your configuration to run the Chrome browser tests when running the Angular CLI commands `ng test` and `ng e2e`.
+
+There are configuration files for both the [Karma JavaScript test runner](https://karma-runner.github.io/latest/config/configuration-file.html)
+and [Protractor](https://www.protractortest.org/#/api-overview) end-to-end testing tool,
+which you must adjust to start Chrome without sandboxing.
```
Comment (jelbourn) [inline aio/content/guide/continuous-integration.md:91]:
```suggestion
An Angular project typically contains configuration files for both unit tests (with the [Karma test runner](https://karma-runner.github.io/latest/config/configuration-file.html)) and end-to-end tests (with [Protractor](https://www.protractortest.org/#/api-overview)). You can control the browser in which
your tests run by modifying these files as below. These configurations include the `--headless` flag to run Chrome without bringing up the browser's full user interface.
```

The `--no-sandbox` bit here is tricky. The Chrome team pretty strong discourages use of this flag because it's a potential security pitfall. However, omitting this flag would mean that the user has to do additional setup with their container image, and that depends on the environment. There's also official articles for [Headless Chrome](https://developers.google.com/web/updates/2017/04/headless-chrome) and [Automated testing with headless Chrome](https://developers.google.com/web/updates/2017/06/headless-karma-mocha-chai), but that content wouldn't cover Protractor, which is an Angular-specific thing. 

So... maybe something like
```markdown
These examples also use the `--no-sandbox` flag, which
[disables Chrome's sandboxing feature](https://developers.google.com/web/tools/puppeteer/troubleshooting#setting_up_chrome_linux_sandbox). This flag should only be used when testing code that you *completely trust*.
The setup to run headless Chrome to run with sandboxing enabled will vary based
on your CI environment.
```

File: aio/content/guide/testing.md [hunk 13]
```diff
@@ -1,49 +1,16 @@
...
-{@a setup}
-
-## Set up testing
+# Testing your application
 
 The Angular CLI downloads and installs everything you need to test an Angular application with the [Jasmine test framework](https://jasmine.github.io/).
 
-The project you create with the CLI is immediately ready to test.
-Just run the [`ng test`](cli/test) CLI command:
+To test an application created using the CLI command [`ng new`](cli/new), run the [`ng test`](cli/test) command:
```
Comment (jelbourn) [inline aio/content/guide/testing.md:5]: It feels a little roundabout to say 
> an application created using the CLI command `ng new`

versus 

>an Angular CLI project

File: aio/content/guide/testing.md [hunk 14]
```diff
@@ -54,240 +21,65 @@ The console output looks a bit like this:
...
+<header>Using other test frameworks</header>
 
-The CLI generated a test file for the `AppComponent` named `app.component.spec.ts`.
-
-<div class="alert is-important">
-
-The test file extension **must be `.spec.ts`** so that tooling can identify it as a file with tests (AKA, a _spec_ file).
+Though the CLI uses the Jasmine test framework by default, you can unit test an Angular application with other testing libraries and test runners.
+This guide specifically covers Jasmine and Karma.
+If you use another option, be sure to research its distinctive installation procedures, configuration, and syntax.
```
Comment (jelbourn) [inline aio/content/guide/testing.md:52]:
```suggestion
By default, Angular CLI uses the Jasmine test framework and the Karma test runner. While this
guide only covers this default setup, you can change your project to use other test frameworks
and test runners.
```
I might omit the last sentence here since it feels a bit awkward.

File: aio/content/guide/testing.md [hunk 15]
```diff
@@ -54,240 +21,65 @@ The console output looks a bit like this:
...
-This article explains how to configure your project to run Circle CI and Travis CI, and also update your test configuration to be able to run tests in the Chrome browser in either environment.
-
+{@a test-name-location}
 
-### Configure project for Circle CI
+## Finding test files
 
-Step 1: Create a folder called `.circleci` at the project root.
+The CLI includes test files when it generates a new Angular application.
+The testing framework identifies test files, or "spec files", by the extension `spec.ts`.
```
Comment (jelbourn) [inline aio/content/guide/testing.md:61]:
```suggestion
The testing framework identifies test files, or "spec" files, by the extension `spec.ts`.
```
Would the quotes just be around "spec" here since files parallels the first part of the sentence?

File: aio/content/guide/continuous-integration.md [hunk 16]
```diff
@@ -0,0 +1,137 @@
...
+1. Commit your changes and push them to your repository.
+
+1. [Sign up for Travis CI](https://travis-ci.org) and [add your project](https://travis-ci.org/getting_started).
+To trigger a build, push a new commit or use the Travis dashboard UI.
+
+For more information about Travis CI testing, see the [Travis CI documentation](https://docs.travis-ci.com/).
+
+## Configuring the CLI for CI testing in Chrome
+
+You can  adjust your configuration to run the Chrome browser tests when running the Angular CLI commands `ng test` and `ng e2e`.
```
[the remark this answers is not in the document]
Comment (LayZeeDK) [reply aio/content/guide/continuous-integration.md:89]: I suspect you're right. We haven't explained headless mode in this section, as far as I can tell.

[the remark this answers is not in the document]
Comment (kyliau) [reply aio/content/guide/continuous-integration.md:89]: I'd rephrase this as:
> Angular CLI is configured to run Chrome out of the box, but such configuration usually does not work on CI due to the lack of a windowing system. For automated testing environment, we need to update the CLI configuration to run the Chrome browser in a [headless](https://developers.google.com/web/updates/2017/04/headless-chrome) environment.

[the remark this answers is not in the document]
Comment (LayZeeDK) [reply aio/content/guide/continuous-integration.md:89]: >For automated testing environment**s** (...)

[the remark this answers is not in the document]
Comment (jelbourn) [reply aio/content/guide/continuous-integration.md:89]: +1, that sounds great to me
